File: app/src/main/java/me/devsaki/hentoid/activities/sources/PornPicsActivity.java
package me.devsaki.hentoid.activities.sources;

import me.devsaki.hentoid.enums.Site;

public class PornPicsActivity extends BaseWebActivity {

    private static final String DOMAIN_FILTER = "pornpics.com";
    private static final String GALLERY_FILTER = "/galleries/";

    Site getStartSite() {
        return Site.PORNPICS;
    }

    @Override
    boolean allowMixedContent() {
        return false;
    }


    @Override
    protected CustomWebViewClient getWebClient() {
        CustomWebViewClient client = new CustomWebViewClient(GALLERY_FILTER, this);
        client.restrictTo(DOMAIN_FILTER);
        return client;
    }
}

File: app/src/main/java/me/devsaki/hentoid/activities/sources/PornPicsActivity.kt
package me.devsaki.hentoid.activities.sources

import me.devsaki.hentoid.enums.Site

private const val DOMAIN_FILTER = "pornpics.com"
private val GALLERY_FILTER = arrayOf("/galleries/[\\w-]+/$")

class PornPicsActivity : BaseBrowserActivity() {
    override fun getStartSite(): Site {
        return Site.PORNPICS
    }

    override fun allowMixedContent(): Boolean {
        return false
    }


    override fun createWebClient(): CustomWebViewClient {
        val client = CustomWebViewClient(getStartSite(), GALLERY_FILTER, this)
        client.restrictTo(DOMAIN_FILTER)
        return client
    }
}